lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -0,0 +1,236 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <at_cmd.h>
+#include <at_cmd_parser/at_cmd_parser.h>
+#include <at_cmd_parser/at_params.h>
+#include <bsd_limits.h>
+#include <modem_key_mgmt.h>
+#include <stdio.h>
+#include <string.h>
+#include <zephyr.h>
+
+#define MODEM_KEY_MGMT_OP_LS "AT%CMNG=1"
+#define MODEM_KEY_MGMT_OP_RD "AT%CMNG=2"
+#define MODEM_KEY_MGMT_OP_WR "AT%CMNG=0"
+#define MODEM_KEY_MGMT_OP_RM "AT%CMNG=3"
+
+#define AT_CMNG_PARAMS_COUNT 5
+#define AT_CMNG_CONTENT_INDEX 4
+
+static char scratch_buf[4096];
+static char at_cmee_strings[2][10] = {"AT+CMEE=0", "AT+CMEE=1"};
+
+static int cmee_active(void)
+{
+	int result;
+	char response[sizeof("+CMEE: X\r\n") + 1];
+
+	memset(response, 0, sizeof(response));
+
+	result = at_cmd_write("AT+CMEE?", response, sizeof(response), NULL);
+	if ((result == 0) && strlen(response) > 0) {
```

Comment:
Why do you check the length of the response here?
Or rather, can you describe in which case a response can have length zero?

Reply:
Please don't check for success, check for error and return.

Reply:
I believe checking for error makes it more imprecise, as errors are given as both negative and positive error codes from `at_cmd_write`. I can probably remove the check for response content and trust that the `at_cmd_write` did not got a "OK" without any payload (which was my case).

Reply:
Alternate in direction you propose would be:
```
if ((error < 0) || ((error > 0) && (state == AT_CMD_ERROR_CME)))
```
If we go for this, i believe we would also have modify the call to `at_cmd_write` to pass in the `state` param, so that we can figure out whether this error is on the positive or negative side. Hence, why i went for success checking here.

Reply:
For now, i have removed the response buffer size check.